### PR TITLE
MGDAPI-231_Reconcile_TopologySpreadConstraints_to_3scale

### PR DESCRIPTION
**Pull request by CathalOConnorRH**
Reconcile TopologySpreadConstraints to 3scale

# Description

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: pkg/products/threescale/reconciler.go [hunk 1]
```diff
@@ -1551,3 +1575,23 @@ func (r *Reconciler) deleteConsoleLink(ctx context.Context, serverClient k8sclie
...
+
+	ns := global.NamespacePrefix + defaultInstallationNamespace
+
+	for _, name := range threeScaleDeploymentConfigs {
+		deploymentConfig := &appsv1.DeploymentConfig{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: ns}}
+		err := serverClient.Get(ctx, k8sclient.ObjectKey{Name: name, Namespace: ns}, deploymentConfig)
+
+		if err != nil {
+			logrus.Errorf("failed to get DeploymentConfig %s in namespace %s with error: %s", name, ns, err)
+		}
```
Comment (sergioifg94) [inline pkg/products/threescale/reconciler.go:1589]:
```suggestion
```

There's no need for this block of code as the `ReconcileZoneTopologySpreadConstraints` function will retrieve the DC anyways:
https://github.com/integr8ly/integreatly-operator/pull/1219/files#diff-618cf8899a2bdb3f5ff975a8867ba9a5R26-R33

**Review by sergioifg94 — CHANGES_REQUESTED** [review]
Code looks good. Just that one suggestion

File: pkg/products/threescale/reconciler.go (hunk 1 above)
Comment (CathalOConnorRH) [reply pkg/products/threescale/reconciler.go:1589]: Ah great. thanks. I've updated now.

File: pkg/products/threescale/reconciler.go [hunk 2]
```diff
@@ -1551,3 +1575,18 @@ func (r *Reconciler) deleteConsoleLink(ctx context.Context, serverClient k8sclie
...
+	ns := global.NamespacePrefix + defaultInstallationNamespace
+
+	for _, name := range threeScaleDeploymentConfigs {
+		deploymentConfig := &appsv1.DeploymentConfig{}
+
+		_, err := resources.ReconcileZoneTopologySpreadConstraints(ctx, serverClient, k8sclient.ObjectKey{Name: name, Namespace: ns}, resources.SelectFromDeploymentConfig, "app", deploymentConfig)
+		if err != nil {
+			return integreatlyv1alpha1.PhaseInProgress, err
+		}
+	}
```
Comment (briangallagher) [inline pkg/products/threescale/reconciler.go:1590]: @CathalOConnorRH , I think you might need an "PhaseAwaitingComponents" check here

**Comment by briangallagher** [comment]
@CathalOConnorRH Confirmed, just the one comment above

File: pkg/products/threescale/reconciler.go (hunk 2 above)
Comment (CathalOConnorRH) [reply pkg/products/threescale/reconciler.go:1590]: Thanks Brian. I've updated now.

**Pull request merged**